packages/flutter_tools/lib/src/desktop_device.dart

```diff
@@ -214,3 +223,34 @@ class DesktopLogReader extends DeviceLogReader {
     // Nothing to dispose.
   }
 }
+
+/// An implementation of a devFS writer which copies physical files for devices
+/// running on the same host.
+class DesktopDevFSWriter implements DevFSWriter {
```

Comment:
We also discussed using this same code for the iOS simulator (can be in a future PR).